src/ucp/rndv/proto_rndv.inl

```diff
@@ -133,6 +130,44 @@ ucp_proto_rndv_rkey_destroy(ucp_request_t *req)
 #endif
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t ucp_proto_rndv_frag_request_alloc(
+        ucp_worker_h worker, ucp_request_t *req, ucp_request_t **freq_p)
+{
+    ucp_request_t *freq;
+
+    freq = ucp_request_get(worker);
+    if (freq == NULL) {
+        ucs_error("failed to allocated rendezvous send fragment");
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    ucp_trace_req(req, "allocated rndv fragment %p", freq);
+    ucp_request_set_super(freq, req);
+    freq->flags   = UCP_REQUEST_FLAG_RNDV_FRAG;
+    freq->send.ep = req->send.ep;
```

Comment:
maybe also fill some `freq` fields, like `req->send.rndv.remote_req_id` or `rkey` here?

Reply:
ucp_proto_rndv_frag_request_alloc() is also going to be used for 3-stage pipeline sender side when allocating fragment for "put" operation, and then these fields will be filled by ucp_proto_rndv_send_start()